### PR TITLE
fix: broken link in contributing guide 

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Fix the wrong link to style guide in `CONTRIBUTING.md`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ GreptimeDB uses the [Apache 2.0 license](https://github.com/GreptimeTeam/greptim
 
 - To ensure that community is free and confident in its ability to use your contributions, please sign the Contributor License Agreement (CLA) which will be incorporated in the pull request process.
 - Make sure all files have proper license header (running `docker run --rm -v $(pwd):/github/workspace ghcr.io/korandoru/hawkeye-native:v3 format` from the project root).
-- Make sure all your codes are formatted and follow the [coding style](https://pingcap.github.io/style-guide/rust/) and [style guide](http://github.com/greptimeTeam/docs/style-guide.md).
+- Make sure all your codes are formatted and follow the [coding style](https://pingcap.github.io/style-guide/rust/) and [style guide](https://github.com/GreptimeTeam/greptimedb/blob/main/docs/style-guide.md).
```
Comment (tisonkun) [inline CONTRIBUTING.md:53]:
```suggestion
- Make sure all your codes are formatted and follow the [coding style](https://pingcap.github.io/style-guide/rust/) and [style guide](docs/style-guide.md).
```

Users always get sources together?

File: docs/style-guide.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ It's mainly an complement to the [Rust Style Guide](https://pingcap.github.io/st
 
 ## Formatting
 
-- Place all `mod` declaration before any `use`.
+- Place all `mod` declaration before any `use` (except the test mod).
```
Comment (tisonkun) [inline docs/style-guide.md:15]: I don't think we need this statement; it seems to cause more confusion (at least I don't get the point before read the PR description).

```
mod tests {
  ...
}
```

is mod + content, not bare "`mod` declarations".

Comment (waynexia) [reply docs/style-guide.md:15]: Okay, as we also have a clippy [lint](https://rust-lang.github.io/rust-clippy/master/index.html#/items_after_test_module) to check this. I'll remove it

**Review by tisonkun — APPROVED** [review]

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**